### PR TITLE
[REF] web: convert load state and push state tests to HOOT

### PR DESCRIPTION
**Pull request by mcm-odoo**
task-3705027

### CODE REVIEW

File: addons/web/static/tests/webclient/actions/load_state.test.js [hunk 1]
```diff
@@ -0,0 +1,1708 @@
...
+import { registry } from "@web/core/registry";
+import { WebClient } from "@web/webclient/webclient";
+import { router, startRouter } from "@web/core/browser/router";
+import { redirect } from "@web/core/utils/urls";
+import { ControlPanel } from "@web/search/control_panel/control_panel";
+import { _t } from "@web/core/l10n/translation";
+import { user } from "@web/core/user";
+import { queryAllTexts } from "@odoo/hoot-dom";
+
+describe.current.tags("headless");
```
Comment (aab-odoo) [inline addons/web/static/tests/webclient/actions/load_state.test.js:31]: I wouldn't tag them headless, there're not as they mount stuff in the dom. If you don't want them to be executed in mobile, tag them "desktop" instead...

File: addons/web/static/tests/webclient/actions/load_state.test.js [hunk 2]
```diff
@@ -0,0 +1,1708 @@
...
+    },
+    {
+        id: 1099,
+        xml_id: "wowl.client_action",
+        tag: "__test__client__action__",
+        target: "main",
+        type: "ir.actions.client",
+        params: { description: "xmlId" },
+    },
+]);
```
Comment (aab-odoo) [inline addons/web/static/tests/webclient/actions/load_state.test.js:148]: do you need all these? For the other suites, if an action is only necessary in a test, I defined it in that test only (typically those actions are really specific).

File: addons/web/static/tests/webclient/actions/load_state.test.js [hunk 3]
```diff
@@ -0,0 +1,1708 @@
...
+                <header></header>
+                <sheet>
+                    <div class="oe_button_box" name="button_box">
+                        <button class="oe_stat_button" type="action" name="1" icon="fa-star" context="{'default_partner': id}">
+                            <field string="Children" name="child_ids" widget="statinfo"/>
+                        </button>
+                    </div>
+                    <field name="display_name"/>
+                </sheet>
+            </form>
```
Comment (aab-odoo) [inline addons/web/static/tests/webclient/actions/load_state.test.js:219]: same here, do you need all those views in several tests?

File: addons/web/static/tests/webclient/actions/push_state.test.js [hunk 4]
```diff
@@ -0,0 +1,602 @@
...
+ * @param {{ env: import("@web/env").OdooEnv }} [options]
+ */
+async function mountWebClient(options) {
+    await mountWithCleanup(WebClient, options);
+    // Wait for visual changes caused by a potential loadState
+    await animationFrame();
+    // wait for BlankComponent
+    await animationFrame();
+    // wait for the regular rendering
+    await animationFrame();
```
Comment (aab-odoo) [inline addons/web/static/tests/webclient/actions/push_state.test.js:196]: I did not need such an helper in the other suites, are you sure you need it?

Comment (mcm-odoo) [reply addons/web/static/tests/webclient/actions/push_state.test.js:196]: Indeed, it's not needed for these tests but needed for load state

File: addons/web/static/tests/webclient/actions/load_state.test.js (hunk 2 above)
Comment (mcm-odoo) [reply addons/web/static/tests/webclient/actions/load_state.test.js:148]: Yes, it's already a subset of [this list](https://github.com/odoo/odoo/blob/master/addons/web/static/tests/legacy/webclient/helpers.js#L151). Tests are duplicated in this suite with one that checks old url and other the new url.

**Comment by mcm-odoo** [comment]
@robodoo delegate=juliusc2066

**Comment by aab-odoo** [comment]
robodoo delegate=juliusc2066

**Comment by Polymorphe57** [comment]
@robodoo rebase-ff r+

**Pull request closed without merging**